src/sparsezoo/analyze_v2/parameter_analysis.py

```diff
@@ -0,0 +1,214 @@
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Union
+
+import numpy
+import yaml
+from onnx import NodeProto
+
+from sparsezoo.analyze_v2.model_validator import (
+    DistributionAnalysisModel,
+    ParameterAnalysisModel,
+    QuantizationAnalysisModel,
+    SparsityAnalysisModel,
+)
+from sparsezoo.utils import (
+    ONNXGraph,
+    get_node_num_four_block_zeros_and_size,
+    get_node_param_counts,
+    get_node_weight,
+    get_node_weight_bits,
+    get_numpy_distribution_statistics,
+    get_numpy_entropy,
+    get_numpy_modes,
+    get_numpy_percentiles,
+    is_quantized_layer,
+)
+
+
+class ParameterAnalysis:
+    """
+    Given model_graph and node, compute
+
+    1. counts / sparsity level
+    2. bits / quantization level
+    3. distribution
+
+    with respect to the weights in the node, if any.
+    """
+
+    def __init__(
+        self,
+        model_graph: ONNXGraph,
+        node: NodeProto,
+    ):
+        self.model_graph = model_graph
+        self.node = node
+
+        self.counts: Dict[str, int] = {}  # single grouping param counts
+        self.bits: Dict[str, int] = {}  # Tensor grouping bits
+
+        self.sparsity_analysis_model = self.get_sparsity()
+        self.quantization_analysis_model = self.get_quantization()
+        self.distribution_model = self.get_distribution()
+
+    def get_sparsity(self) -> List["SparsityAnalysisModel"]:
+        """Get the number of dense and sparse weights"""
+
+        data = get_parameter_counts(self.model_graph, self.node)
+        sparsity_analysis_model = []
+        for grouping, counts_dict in data.items():
+            if grouping == "single":
+                self.counts = counts_dict
+
+            sparsity_analysis_model.append(
+                SparsityAnalysisModel(grouping=grouping, **counts_dict)
+            )
+
+        return sparsity_analysis_model
+
+    def get_quantization(self) -> List["QuantizationAnalysisModel"]:
+        """Get the number of bits and quantized bits from weights"""
+        data = get_parameter_bits(self.model_graph, self.node)
+        quantization_analysis_model = []
+        for grouping, counts_dict in data.items():
+            if grouping == "tensor":
+                self.bits = counts_dict
+
+            quantization_analysis_model.append(
+                QuantizationAnalysisModel(grouping=grouping, **counts_dict)
+            )
+
+        return quantization_analysis_model
+
+    def get_distribution(self) -> "DistributionAnalysisModel":
+        """Get the distribution statistics with respect to the weights"""
+        distribution_dct = get_parameter_distribution(self.model_graph, self.node)
+        return DistributionAnalysisModel(**distribution_dct)
```

Comment:
why not just return `None` if there's no weight. I think this will make it easier to check, rather than a user getting a constructed object with all the attrs set to None